### PR TITLE
fix http proxy on mariner nodes

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_main.sh [hunk 1]
```diff
@@ -65,7 +65,11 @@ fi
 
 if [[ "${SHOULD_CONFIGURE_HTTP_PROXY}" == "true" ]]; then
     if [[ "${SHOULD_CONFIGURE_HTTP_PROXY_CA}" == "true" ]]; then
-        configureHTTPProxyCA || exit $ERR_UPDATE_CA_CERTS
+        if [[ $OS == $MARINER_OS_NAME ]]; then
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_main.sh:68]: can we move this inside the func and avoid pivoting on the positional params? feels like recipe for error in ordering or similar

Comment (cameronmeissner) [reply parts/linux/cloud-init/artifacts/cse_main.sh:68]: yeees, was actually thinking the same :)

**Review by alexeldeib — APPROVED** [review]
@hbeberman mind taking a look?

**Comment by cameronmeissner** [comment]
just verified with passing HTTP proxy e2e's

**Review by hbeberman — APPROVED** [review]
LGTM :shipit:

**Pull request merged**